### ISSUE TITLE
clarification: what does language structure in the ablation study in the paper mean?

### ISSUE THREAD

**Issue by Edwardmark** [mmaaz60/mvits_for_class_agnostic_od#29]
Thans for your great work and code, I have some question to clarify.
(1)what does language structure in the ablation study in the paper mean? How do you achieve this?
for example, if the caption is "a cat in the chair", how to remove the caption and keep the language structure, what is the input and the losses used in this setting?
![image](https://github.com/mmaaz60/mvits_for_class_agnostic_od/assets/15247551/df3f381a-0f55-40b7-b4aa-04f59eed1063)
could you give me an example? What is the input to the model and loss function if you remove all captions?
did you still use soft token prediction loss without language feas concated?

![image](https://github.com/mmaaz60/mvits_for_class_agnostic_od/assets/15247551/25bfc923-939c-4f68-856f-2af6beba5af3)
(2) in Table 9 what is setting1 input and loss function, cound you give me an example or could you give me the code of the ablation study?  Thanks
 @mmaaz60 @hanoonaR  Thanks.
![Uploading image.png…]()

**Comment by mmaaz60** [comment]
Hi @Edwardmark,

Thank you for your interest in our work.

In LMDet dataset (same dataset as in MDETR paper), each image has multiple corresponding captions. Specifically, we have around 118K images and 1.3M corresponding captions. The training is performed such that each epoch contains 1.3M iterations, which means each image is seen multiple times in one epoch but with different caption/context/boxes.

When we say `Language Structure`, we mean the structure defined by image-text pairs, for example the structure where each image is presented to the model with different context. And when we say `removing captions but keeping the language structure intact`, it means that there is no text encoder now, however we still have 1.3M iterations in one epoch. It means one image will be still seen multiple times in one epoch. In this case, the standard detection losses are used (i.e. exactly same as in Deformable DETR paper) and the training code to train this model is available at [MAVL-minus-Language](https://github.com/mmaaz60/mdef_detr/tree/mdef_detr_minus_language) for your reference.

I hope this will help. Please let me know if you have any questions. Thank You.
